client/src/useJobPaginator.js

```diff
@@ -0,0 +1,191 @@
+import { useEffect, useState } from "react";
+import MagicLabeler from "./MagicLabeler";
+import AjaxHelper from "./AjaxHelper";
+import {
+    countMagicLabels,
+    deleteImage,
+    deletePageValidation,
+    getAddedPages,
+    getNoneRemovedPages,
+    getStatus,
+    getJob,
+    getLabel,
+    putJob,
+    validatePublish,
+    validateSave,
+} from "./JobPaginatorState";
+
+const useJobPaginator = (initialCategory, initialJob) => {
+    const [active, setActive] = useState(false);
+    const [currentPage, setCurrentPage] = useState(0);
+    const [zoom, setZoom] = useState(false);
+    const [order, setOrder] = useState([]);
+    const [magicLabelCache, setMagicLabelCache] = useState([]);
+    const [category, setCategory] = useState(initialCategory);
+    const [job, setJob] = useState(initialJob);
+
+    const saveMagicLabels = () => {
+        order.forEach((o, orderIndex) => {
+            if (!getLabel(order, orderIndex)) {
+                setLabel(orderIndex, getMagicLabel(orderIndex));
+            }
+        });
+    };
+
+    const setLabel = (imageNumber, text) => {
+        setMagicLabelCache([]); // clear label cache whenever there is a change
+        if (typeof order[imageNumber] === "undefined") {
+            return;
+        }
+        order[imageNumber].label = text !== null && text.length === 0 ? null : text;
+        setOrder(order);
+        dispatchEvent(new Event("Prep.editted"));
+    };
+
+    const getMagicLabel = (imageNumber) => {
+        const label = getLabel(order, imageNumber);
+
+        if (label) {
+            return label;
+        }
+        if (typeof magicLabelCache[imageNumber] === "undefined") {
+            magicLabelCache[imageNumber] = MagicLabeler.getLabel(imageNumber, getMagicLabel);
+        }
+        setMagicLabelCache(magicLabelCache);
```

Comment:
Is there a reason this was moved out of the if statement above? I would think that we only need to set this value when it changes, but maybe there's a subtlety I'm not aware of.

Reply:
I was trying to fix tsom things. I'm putting it back.